src/libraries/System.IO.FileSystem/tests/File/Move.cs

```diff
@@ -298,10 +284,18 @@ public void UnixWildCharacterPath()
         [Theory,
             MemberData(nameof(ControlWhiteSpace))]
         [PlatformSpecific(TestPlatforms.Windows)]
-        public void WindowsControlPath_Core(string whitespace)
+        public void WindowsControlPath(string whitespace)
         {
             FileInfo testFile = new FileInfo(GetTestFilePath());
-            Assert.ThrowsAny<IOException>(() => Move(testFile.FullName, Path.Combine(TestDirectory, whitespace)));
+            // Same exception message, different exception type
+            if (PlatformDetection.IsNetCore)
+            {
+                Assert.ThrowsAny<IOException>(() => Move(testFile.FullName, Path.Combine(TestDirectory, whitespace)));
+            }
+            else
+            {
+                Assert.Throws<ArgumentException>(() => Move(testFile.FullName, Path.Combine(TestDirectory, whitespace)));
+            }
```

Comment:
Same question about when we choose to use which structure for this kind of test.